### PR TITLE
feat: build wheels using CIBuildWheel

### PR DESCRIPTION
**Pull request by tseaver**

### CODE REVIEW

File: .github/workflows/use-cibuildwheel.yml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        name: python-package-distributions
+        path: ./wheelhouse/google_crc32c*.whl
+
+  build-wheels-macos:
+    name: Build wheels on ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+        - macos-10.15
+       #- macos-11
```
Comment (andrewsg) [inline .github/workflows/use-cibuildwheel.yml:57]: Due to the changes we made previously to fix the OS X 11 bug, we ended up not needing a separate Big Sur build, correct?

File: .github/workflows/use-cibuildwheel.yml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    runs-on: ${{ matrix.os }}
+    steps:
+
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
+
+    - name: Build C Library
```
Comment (andrewsg) [inline .github/workflows/use-cibuildwheel.yml:65]: Why is this needed as a separate github action step if it's also being built as a cibuildwheel prep step?

**Review by andrewsg — COMMENTED** [review]
Shall we change the PR name to feat: so that we get a version bump? I should have added one back in 1.1.3, but better late than never.

File: .github/workflows/use-cibuildwheel.yml (hunk 1 above)
Comment (tseaver) [reply .github/workflows/use-cibuildwheel.yml:57]: I believe so -- even the wheels built on Big Sur were segfaulting on Big Sur with Python 3.8 before that fix.

File: .github/workflows/use-cibuildwheel.yml (hunk 2 above)
Comment (tseaver) [reply .github/workflows/use-cibuildwheel.yml:65]: We only build it as a prep step on Linux, where we need to build it using the same `manylinux` image being used to build the wheels.  Here, splitting it out can help optimize the build (via a cache -- see below where I added the cache for the Windows library build), as well as making the built libraries potentitally available as downloadable artifacts.

**Comment by tseaver** [comment]
@andrewsg 
> Shall we change the PR name to feat: so that we get a version bump? I should have added one back in 1.1.3, but better late than never.

Makes sense -- the M1 builds are certainly a new feature.

One question before we merge this PR:  the workflow I have now makes more sense for the release-tag build than for a presubmit:  it bulids and tests all the wheels we know how to build.  Perhaps we could trim back the wheels we build and test for the presubmit? (No `Win32`, maybe, and maybe only a couple of `manylinux` versions?

File: .github/workflows/use-cibuildwheel.yml [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            echo "::set-output name=arch::x86"
+        else
+            echo "::set-output name=arch::AMD64"
+        fi
+
+    - name: Build Wheels
+      uses: pypa/cibuildwheel@v2.1.1
+      env:
+        CIBW_ARCHS_WINDOWS: ${{ steps.platform-arch.outputs.arch }}
+        CIBW_ENVIRONMENT: CRC32C_PURE_PYTHON="0" CRC32C_INSTALL_PREFIX="D:/a/python-crc32c/python-crc32c/usr"
```
Comment (tseaver) [inline .github/workflows/use-cibuildwheel.yml:173]: I'm not really satisfied that using this hard-wired expansion of the path is best, but the various stabs I made at using `${{ github.workspace }}` to build the path all ended up getting mangled by CIBW passing through backslash-mangled versions.

Comment (andrewsg) [reply .github/workflows/use-cibuildwheel.yml:181]: It's pretty comical that we are still dealing with the same backslash issues after some 40 years of DOS and Windows.

**Comment by andrewsg** [comment]
Normally I'd want to keep the presubmit as close to release as possible. What's the benefit of trimming it back in this case?

**Comment by tseaver** [comment]
@andrewsg 
> Normally I'd want to keep the presubmit as close to release as possible. What's the benefit of trimming it back in this case?

Turnaround time:  building all the wheels is slow, and seems unlikely to uncover any issues.  Not to mention heating the planet. :)

**Comment by andrewsg** [comment]
Okay, if we can cut down on presubmit safely without exposing ourselves to any risk of cutting a release where the build doesn't complete and partial artifacts are pushed out, then I'm fine with that.

**Comment by tseaver** [comment]
@andrewsg I think we can add handling a [`scheduled` event](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events) to the "build everything" workflow, and have it [skip the "push to PyPI" job](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif) in that case.  We can then use the artifacts from the nightly build to verify what would be released.

**Comment by andrewsg** [comment]
Hmm, that sounds a bit complex... we've got quite a bit going on in this release, do you think we should push that through as well or can we accept the longer presubmits for now?

**Review by andrewsg — APPROVED** [review]
